[BUG][kotlin-spring] Gradle build fails when generated with `interfaceOnly=true`

```shell
$ gradle build -x test
> Task :bootJar FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':bootJar'.
> Error while evaluating property 'mainClass' of task ':bootJar'.
   > Failed to calculate the value of task ':bootJar' property 'mainClass'.
      > Main class name has not been configured and it could not be resolved

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
```

This happens when building locally and also on the CI pipeline: https://github.com/OpenAPITools/openapi-generator/actions/runs/9314727937/job/25639472824 – blocking any kotlin-related PRs from succeeding.

@stefankoppier @Rugal any ideas?

Strangely enough, i also tried building against the very first version locally (1e0e8fe0eb), same result.

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

[ci] [kotlin]

--- a/.github/workflows/samples-kotlin-server-jdk17.yaml
+++ b/.github/workflows/samples-kotlin-server-jdk17.yaml
@@ -28,6 +28,7 @@ jobs:
           # server
           - samples/server/petstore/kotlin-springboot-3
           - samples/server/petstore/kotlin-springboot-request
+          - samples/server/petstore/kotlin-springboot-request-cookie
           - samples/server/petstore/kotlin-server/javalin
           # comment out due to gradle build failure
           # - samples/server/petstore/kotlin-spring-default/

--- a/modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradle-sb3-Kts.mustache
+++ b/modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradle-sb3-Kts.mustache
@@ -13,6 +13,12 @@ tasks.withType<KotlinCompile> {
     kotlinOptions.jvmTarget = "17"
 }
 
+{{#interfaceOnly}}
+tasks.bootJar {
+    enabled = false
+}
+
+{{/interfaceOnly}}
 plugins {
     val kotlinVersion = "1.7.10"
     id("org.jetbrains.kotlin.jvm") version kotlinVersion

--- a/modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradleKts.mustache
+++ b/modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradleKts.mustache
@@ -21,9 +21,9 @@ tasks.withType<KotlinCompile> {
 }
 
 {{#interfaceOnly}}
-    tasks.bootJar {
+tasks.bootJar {
     enabled = false
-    }
+}
 
 {{/interfaceOnly}}
 plugins {

--- a/modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradleKts.mustache
+++ b/modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradleKts.mustache
@@ -20,6 +20,12 @@ tasks.withType<KotlinCompile> {
     kotlinOptions.jvmTarget = "1.8"
 }
 
+{{#interfaceOnly}}
+    tasks.bootJar {
+    enabled = false
+    }
+
+{{/interfaceOnly}}
 plugins {
     val kotlinVersion = "1.6.21"
     id("org.jetbrains.kotlin.jvm") version kotlinVersion

--- a/samples/server/petstore/kotlin-springboot-request-cookie/build.gradle.kts
+++ b/samples/server/petstore/kotlin-springboot-request-cookie/build.gradle.kts
@@ -13,6 +13,10 @@ tasks.withType<KotlinCompile> {
     kotlinOptions.jvmTarget = "17"
 }
 
+tasks.bootJar {
+    enabled = false
+}
+
 plugins {
     val kotlinVersion = "1.7.10"
     id("org.jetbrains.kotlin.jvm") version kotlinVersion

--- a/samples/server/petstore/kotlin-springboot-request/build.gradle.kts
+++ b/samples/server/petstore/kotlin-springboot-request/build.gradle.kts
@@ -13,6 +13,10 @@ tasks.withType<KotlinCompile> {
     kotlinOptions.jvmTarget = "17"
 }
 
+tasks.bootJar {
+    enabled = false
+}
+
 plugins {
     val kotlinVersion = "1.7.10"
     id("org.jetbrains.kotlin.jvm") version kotlinVersion

